src/plugins/intel_gna/src/ops/util/util.hpp

```diff
@@ -274,6 +279,115 @@ static bool has_32bit_output(const std::shared_ptr<ngraph::Node>& node) {
 inline bool has_32bit_input(const std::shared_ptr<ngraph::Node>& node) {
     return is_activation(node) || is_pooling(node);
 }
+
+static bool is_op_supported(const std::shared_ptr<ov::Node>& node,
+                            const std::string& gnaCompileTarget,
+                            const GNAPluginNS::Config config,
+                            bool raise_exception = false) {
+    if (ngraph::op::is_parameter(node)) {
+        const std::set<ov::element::Type> supported = { ov::element::u8, ov::element::i16, ov::element::f32 };
+        if (supported.count(node->get_element_type()) == 0) {
+            if (raise_exception) {
+                THROW_GNA_EXCEPTION << "The plugin does not support input precision with " +
+                    std::string(node->get_type_name()) + " format. Supported input precisions f32, i16, u8\n";
+            }
+        }
+        return true;
+    } else if (ngraph::op::is_constant(node)) {
+        const std::set<ov::element::Type> supported = {
```

Comment:
It would be great to define global constants outside of the particular functions.
We can put them in separate header file

Reply:
> It would be great to define global constants outside of the particular functions. We can put them in separate header file

I don't see how we can use this list. My assumption that it will be modified in the future and become short.